src/uio-pruss-interface.cc

```diff
@@ -0,0 +1,88 @@
+/* -*- mode: c++; c-basic-offset: 2; indent-tabs-mode: nil; -*-
+ * (c) 2013, 2014 Henner Zeller <h.zeller@acm.org>
+ *
+ * This file is part of BeagleG. http://github.com/hzeller/beagleg
+ *
+ * BeagleG is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * BeagleG is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with BeagleG.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+// Implementation of the hardware interface of the PRU using the uio_pruss
+// driver from the bone line kernel.
+
+#include "pru-hardware-interface.h"
+
+#include <assert.h>
+#include <errno.h>
+#include <pruss_intc_mapping.h>
+#include <prussdrv.h>
+#include <stdio.h>
+#include <strings.h>
+#include <string.h>
+
+#include "logging.h"
+
+// Generated PRU code from motor-interface-pru.p
+#include "motor-interface-pru_bin.h"
+
+#if PRU_NUM==0
+#define PRU_DATARAM PRUSS0_PRU0_DATARAM
```

Comment:
The 'canoncial' way to do nested defines is to have the # start at the beginning, then some indentation. So something like
```
#if PRU_NUM == 0
#  define PRU_DATARAM ...  // notice the space between '#' and define.
#  define ....
#else
#  define PRU_DATARAM ...
#  define ...
#endif
```